client/basic-authorship-ver/src/basic_authorship.rs

```diff
@@ -426,10 +418,20 @@ where
 	) -> Result<Proposal<Block, backend::TransactionFor<B, Block>, PR::Proof>, sp_blockchain::Error>
 	{
 		let propose_with_start = time::Instant::now();
+		let mut inherent_data = inherent_data.clone();
 
 		let mut block_builder =
 			self.client.new_block_at(&self.parent_id, inherent_digests, PR::ENABLED)?;
 
+		if let Ok(None) = inherent_data
```

Comment:
that should be removed from here and inherent should be injected directly in unit tests, in real scenario the shuffling seed is already in inherent_data